.github/scripts/generate_pytorch_test_matrix.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python3
+
+"""Generates a matrix to be utilized through github actions
+
+Will output a matrix to represent our testing configurations, which is currently
+dictated by just sharding.
+
+"""
+
+import json
+import os
+from typing import List
+
+
+NUM_TEST_SHARDS = int(os.getenv('NUM_TEST_SHARDS', '1'))
+
+def generate_sharding_list() -> List[int]:
+    return list(range(1, NUM_TEST_SHARDS + 1))
+
+
+def main() -> None:
+    print(json.dumps({
+        'test_config': generate_sharding_list()
+    }))
```

Comment:
Let's keep output a bit more human readable
```suggestion
    print(json.dumps(
        {'test_config': generate_sharding_list()},
        sort_keys=True, 
        indent=4
    ))
```

Reply:
So this somehow breaks GHA parsing: https://github.com/pytorch/pytorch/actions/runs/947332678

Reply:
I think the fromJSON function in GHA can't handle the indentation. I'll change it back for now and we can look into it in the future